src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2643,20 +2655,28 @@ internal static bool CheckValEscape(SyntaxNode node, BoundExpression expr, uint
                     var operandExpression = ((BoundPointerIndirectionOperator)expr).Operand;
                     return CheckValEscape(operandExpression.Syntax, operandExpression, escapeFrom, escapeTo, checkingReceiver, diagnostics);
 
+                case BoundKind.AsOperator:
+                case BoundKind.AwaitExpression:
+                case BoundKind.ConditionalAccess:
+                case BoundKind.NullCoalescingOperator:
+                case BoundKind.ArrayAccess:
+                    // only possible in error cases (if possible at all)
+                    return false;
+
                 default:
```

Comment:
> default: [](start = 15, length = 9)

Why are we moving away from throw here? The existing code is finding a lot of bugs.

Reply:
There is a reasonable default behavior for the nodes that we do not want explicitly handle here. The alternative is that any time we have something unexpected here we have crash in VS. 

The quality of the bugs that come from this lately is not very high - we are basically just adding cases to the switch section that does that same default behavior. 

Basically - I think the resulting "improvements" no longer justify the risk of crashes and general customer grief for this case. 
Considering that there is a safe default behavior - the assert seems more appropriate at this point.